src/backend/drm/impl/Atomic.cpp

```diff
@@ -100,8 +100,16 @@ void Aquamarine::CDRMAtomicRequest::addConnector(Hyprutils::Memory::CSharedPoint
     TRACE(backend->log(AQ_LOG_TRACE, std::format("atomic addConnector values: CRTC {}, mode {}", enable ? connector->crtc->id : 0, data.atomic.modeBlob)));
 
     conn = connector;
+    drmModeModeInfo* currentMode = connector->getCurrentMode();
+    bool modeDiffers = true;
+    if (currentMode) {
+        modeDiffers = memcmp(currentMode, &data.modeInfo, sizeof(drmModeModeInfo)) != 0;
+        free(currentMode);
+    }
 
-    addConnectorModeset(connector, data);
+    if (modeDiffers) {
```

Comment:
no {} around short ifs

Reply:
oh sorry, i'm used to rust where you can't omit the braces, will fix